pandas/core/indexes/period.py

```diff
@@ -567,17 +567,32 @@ def get_loc(self, key, method=None, tolerance=None):
         """
 
         if isinstance(key, str):
-            try:
-                return self._get_string_slice(key)
-            except (TypeError, KeyError, ValueError, OverflowError):
-                pass
 
             try:
-                asdt, reso = parse_time_string(key, self.freq)
-                key = asdt
-            except DateParseError:
-                # A string with invalid format
-                raise KeyError(f"Cannot interpret '{key}' as period")
+                loc = self._get_string_slice(key)
+                return loc
+            except (TypeError, ValueError):
+
```

Comment:
can you just pass in the except and out-dent this while block (the _get_string_slice try/except)

Reply:
sure, either way works